x-pack/plugins/actions/server/plugin.ts

```diff
@@ -333,7 +334,12 @@ export class ActionsPlugin implements Plugin<Promise<PluginSetupContract>, Plugi
 
     this.eventLogService!.registerSavedObjectProvider('action', (request) => {
       const client = secureGetActionsClientWithRequest(request);
-      return async (type: string, id: string) => (await client).get({ id });
+      return (objects?: SavedObjectsBulkGetObject[]) =>
+        objects
+          ? Promise.all(
+              objects.map(async (objectItem) => await (await client).get({ id: objectItem.id }))
+            )
+          : new Promise(() => []);
```

Comment:
```suggestion
          : Promise.resolve([])
```

I don't think the `new Promise(() => [])` is doing what we want - it seems to return a promise that never resolves, but I think we want a promise that resolves to an empty array instead.

Same pattern is in the alerts plugin ...

Reply:
Even better, current PR can close the opened issue #70856 and the alerting related work will be a separate PR